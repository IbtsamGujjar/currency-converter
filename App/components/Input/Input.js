import React from 'react';
import PropTypes from 'prop-types';
import {
  View,
  TouchableOpacity,
  Text,
  TextInput,
  StyleSheet,
} from 'react-native';

import { Colors } from '../../../constants/colors';

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.white,
    marginVertical: 10,
    marginHorizontal: 20,
    flexDirection: 'row',
    borderRadius: 5,
  },
  button: {
    padding: 15,
    borderRightColor: Colors.border,
    borderRightWidth: 1,
    backgroundColor: Colors.white,
    borderTopLeftRadius: 5,
    borderBottomLeftRadius: 5,
  },
  buttonText: {
    fontSize: 18,
    color: Colors.grey,
    fontWeight: 'bold',
  },
  input: {
    flex: 1,
    padding: 10,
    fontSize: 16,
    color: Colors.textLight,
  },
});

export const Input = ({
  value,
  buttonText,
  keyboardType,
  onPress,
  onChangeText,
}) => {
  return (
    <View style={styles.container}>
      <TouchableOpacity style={styles.button} onPress={onPress}>
        <Text style={styles.buttonText}>{buttonText}</Text>
      </TouchableOpacity>
      <TextInput
        style={styles.input}
        keyboardType={keyboardType}
        onChangeText={onChangeText}
        value={value}
      />
    </View>
  );
};

Input.defaultProps = {
  value: '',
  buttonText: 'USD',
  keyboardType: 'number-pad',
  onPress: () => {},
  onChangeText: () => {},
};

Input.propTypes = {
  value: PropTypes.string,
  keyboardType: PropTypes.string,
  buttonText: PropTypes.string,
  onPress: PropTypes.func,
  onChangeText: PropTypes.func,
};
