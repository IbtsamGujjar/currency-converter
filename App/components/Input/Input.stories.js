import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { action } from '@storybook/addon-actions';

import { Input } from './Input';
import { CenterView } from '../../../storybook/stories/decorator';

storiesOf('Input', module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add('Editable', () => (
    <Input
      buttonText="USD"
      value="100"
      editable
      onPress={action('tapped-default')}
    />
  ))
  .add('Disabled', () => (
    <Input
      buttonText="USD"
      value="100"
      editable={false}
      onPress={action('tapped-default')}
    />
  ));
