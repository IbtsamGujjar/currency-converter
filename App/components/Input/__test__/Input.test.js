import React from 'react';
import renderer from 'react-test-renderer';
import { Input } from '../Input';

test('Input Snapshot', () => {
  const snapshot = renderer.create(<Input />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
