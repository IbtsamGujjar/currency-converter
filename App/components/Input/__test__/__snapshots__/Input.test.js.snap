// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderRadius": 5,
      "flexDirection": "row",
      "marginHorizontal": 20,
      "marginVertical": 10,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderBottomLeftRadius": 5,
        "borderRightColor": "#e2e2e2",
        "borderRightWidth": 1,
        "borderTopLeftRadius": 5,
        "opacity": 1,
        "padding": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#4f6d7a",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      USD
    </Text>
  </View>
  <TextInput
    allowFontScaling={true}
    editable={true}
    keyboardType="number-pad"
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Object {
        "color": "#797979",
        "flex": 1,
        "fontSize": 16,
        "padding": 10,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
